packages/wonder-blocks-dropdown/src/components/multi-select.tsx

```diff
@@ -217,74 +190,77 @@ type State = Readonly<{
  * </MultiSelect>
  * ```
  */
-export default class MultiSelect extends React.Component<Props, State> {
-    labels: Labels;
-
-    static defaultProps: DefaultProps = {
-        alignment: "left",
-        disabled: false,
-        error: false,
-        light: false,
-        shortcuts: false,
-        selectedValues: [],
-    };
-
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            open: false,
-            searchText: "",
-            lastSelectedValues: [],
-            // merge custom labels with the default ones
-            labels: {...defaultLabels, ...props.labels},
-        };
-        // merge custom labels with the default ones
-        this.labels = {...defaultLabels, ...props.labels};
-    }
-
-    /**
-     * Used to sync the `opened` state when this component acts as a controlled
-     * component
-     */
-    static getDerivedStateFromProps(
-        props: Props,
-        state: State,
-    ): Partial<State> | null {
-        return {
+const MultiSelect = (props: Props) => {
+    const {
+        id,
+        light = false,
+        opener,
+        testId,
+        alignment = "left",
+        dropdownStyle,
+        implicitAllEnabled,
+        isFilterable,
+        labels: propLabels,
+        onChange,
+        onToggle,
+        opened,
+        selectedValues = [],
+        shortcuts = false,
+        style,
+        className,
+        "aria-invalid": ariaInvalid,
+        "aria-required": ariaRequired,
+        disabled = false,
+        error = false,
+        children,
+        dropdownId,
+        ...sharedProps
+    } = props;
+
+    // Merge custom labels with the default ones
+    const labels = {...defaultLabels, ...propLabels};
+
+    // Whether or not the dropdown is open.
+    const [open, setOpen] = React.useState(false);
+
+    // The text input to filter the items by their label. Defaults to an empty
+    // string.
+    const [searchText, setSearchText] = React.useState("");
+
+    // The selected values that are set when the dropdown is opened. We use this
+    // to move the selected items to the top when the dropdown is re-opened.
+    const [lastSelectedValues, setLastSelectedValues] = React.useState<
+        string[]
+    >([]);
+
+    // The DOM reference to the opener element. This is mainly used to set focus
+    // to this element, and also to pass the reference to Popper.js.
+    const [openerElement, setOpenerElement] = React.useState<HTMLElement>();
+
+    React.useEffect(() => {
+        /**
+         * Used to sync the `opened` state when this component acts as a controlled
+         * component
+         */
+        if (disabled) {
             // open should always be false if select is disabled
-            open: props.disabled
-                ? false
-                : typeof props.opened === "boolean"
-                ? props.opened
-                : state.open,
```

Comment:
The logic for `getDerivedStateFromProps` is translated to a useEffect:
- If `disabled` prop is set to `true`, we update the local `open` state to `false`.
- If the `opened` prop changes, we update the local `open` state to the value of the prop
- Otherwise, don't change local `open` state

https://github.com/Khan/wonder-blocks/blob/32b4f84272bf6c7b98e904fb9a87260f6ae467fd/packages/wonder-blocks-dropdown/src/components/multi-select.tsx#L240-L248